src/lib.rs

```diff
@@ -58,59 +68,186 @@
+static MENU_CHANNEL: Lazy<(Sender<MenuEvent>, MenuEventReceiver)> = Lazy::new(unbounded);
+
+/// Gets a reference to the event channel's [MenuEventReceiver]
 /// which can be used to listen for menu events.
-pub fn menu_event_receiver<'a>() -> &'a Receiver<MenuEvent> {
+pub fn menu_event_receiver<'a>() -> &'a MenuEventReceiver {
     &MENU_CHANNEL.1
 }
 
 /// Describes a menu event emitted when a menu item is activated
+#[derive(Debug)]
 pub struct MenuEvent {
     /// Id of the menu item which triggered this event
-    pub id: u64,
+    pub id: u32,
 }
 
-/// This is the root menu type to which you can add
-/// more submenus and later be add to the top of a window (on Windows and Linux)
-/// or used as the menubar menu (on macOS) or displayed as a popup menu.
-///
-/// # Example
-///
-/// ```no_run
-/// let mut menu = muda::Menu::new();
-/// let file_menu = menu.add_submenu("File", true);
-/// let edit_menu = menu.add_submenu("Edit", true);
-/// ```
+/// A root menu that can be added to a Window on Windows and Linux
+/// and used as the app global menu on macOS.
 #[derive(Clone)]
 pub struct Menu(platform_impl::Menu);
 
+impl Default for Menu {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl Menu {
-    /// Creates a new root menu.
+    /// Creates a new menu.
     pub fn new() -> Self {
         Self(platform_impl::Menu::new())
     }
 
-    /// Creates a new [`Submenu`] whithin this menu.
+    /// Creates a new menu with given `items`. It calls [`Menu::new`] and [`Menu::append_items`] internally.
+    pub fn with_items(items: &[&dyn MenuItemExt]) -> Self {
+        let menu = Self::new();
+        menu.append_items(items);
+        menu
+    }
+
+    /// Add a menu item to the end of this menu.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn append(&self, item: &dyn MenuItemExt) {
+        self.0.append(item)
+    }
+
+    /// Add menu items to the end of this menu. It calls [`Menu::append`] in a loop.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn append_items(&self, items: &[&dyn MenuItemExt]) {
+        for item in items {
+            self.append(*item);
+        }
+    }
+
+    /// Add a menu item to the beginning of this menu.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn prepend(&self, item: &dyn MenuItemExt) {
+        self.0.prepend(item)
+    }
+
+    /// Add menu items to the beginning of this menu.
+    /// It calls [`Menu::prepend`] on the first element and
+    /// passes the rest to [`Menu::insert_items`] with position of `1`.
+    ///
+    /// ## Platform-spcific:
+    ///
+    /// - **macOS:** Only [`Submenu`] can be added to the menu
+    pub fn prepend_items(&self, items: &[&dyn MenuItemExt]) {
+        self.prepend(items[0]);
+        self.insert_items(&items[1..], 1);
```

Comment:
Can't we do `insert_items` with a position of `0`, instead of special-casing the first item?

Reply:
yeah you're right

Reply:
yeah you're right